cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -59,6 +63,74 @@ private ModelUtils() {}
     public static final String KUBERNETES_SERVICE_DNS_DOMAIN =
             System.getenv().getOrDefault("KUBERNETES_SERVICE_DNS_DOMAIN", "cluster.local");
 
+    /**
+     * Generates the DNS name of the pod including the cluster suffix
+     * (i.e. usually with the cluster.local - but can be different on different clusters)
+     * Example: my-pod-1.my-service.my-ns.svc.cluster.local
+     *
+     * @param namespace     Namespace of the pod
+     * @param serviceName   Name of the cluster
+     * @param podName       Name of the pod within the STS
```

Comment:
could be not necessarily an STS right?

Reply:
I don't _think_ you can address a pod from a deployment or job like this, and for another matter I _think_ you probably shouldn't in general.

When contacting a deployment, you would normally resolve the service address and allow that to route you to wherever kube deems best (normally round robin).

I think that you would only ever contact a pod like this via an STS due to the statefulness and ordinals, but I might be wrong on that.

Reply:
Also thanks for the approve!

Reply:
Though I am in two minds since even if it is only usable by sts, the helper is at the generic pod level...

I will change it and doc a note in the javadoc

Reply:
Well I think that you could potentially addressing the pod. The problem is that the pod name from a deployment is built from the deployment name, a random number for the underlying replicaset and another random number related to the pod itself. With StatefulSets of course it is simpler due to the naming convention with ordinals. Anyway I agree, this utils class is at pod level so it's better don't mention St's. Anyway it's really a minor thing. The PR LGTM ;)

Reply:
I don't think this has to be STS. But it has to be headless service and it is usually used with STS. The whole idea of stable DNS name for Deploment pods is kinds invalid.